packages/lodestar/src/db/seenSyncCommitteeCache.ts

```diff
@@ -0,0 +1,164 @@
+import bls, {Signature} from "@chainsafe/bls";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {altair, phase0} from "@chainsafe/lodestar-types";
+import {intDiv} from "@chainsafe/lodestar-utils";
+import {List} from "@chainsafe/ssz";
+import {NUM_SLOTS_IN_CACHE, slotRootKey, SlotRootKey} from "./repositories/utils/syncCommittee";
+
+/**
+ * Preaggregate SyncCommitteeSignature into SyncCommitteeContribution
+ * and cache seen SyncCommitteeSignature by slot + validator index.
+ * This stays in-memory and should be pruned per slot.
+ */
+export class SeenSyncCommitteeCache {
+  private readonly config: IBeaconConfig;
+
+  /**
+   * Each array item is respective to a subCommitteeIndex.
+   * Preaggregate into SyncCommitteeContribution.
+   * */
+  private contributionCaches: Map<phase0.Slot, Map<SlotRootKey, altair.SyncCommitteeContribution>>[];
+
+  /**
+   * Each array item is respective to a subCommitteeIndex.
+   * A seen SyncCommitteeSignature is decided by slot + validator index.
+   */
+  private seenCaches: Map<phase0.Slot, Set<phase0.ValidatorIndex>>[];
```

Comment:
If I understand, here we need to know if we have seen a signature for a combination of slot + validatorIndex + subnet. Would it make things simpler just to have a single 
```ts
type SubnetValidatorIndex = string; // `${subnet}-${validatorIndex}`
Map<phase0.Slot, Set<SubnetValidatorIndex>>
```

Reply:
Doing it this way would be consistent with the seenCache of `SeenSyncCommitteeContributionCache`